drivers/npmx/shell/buck.c

```diff
@@ -90,44 +113,59 @@ static int buck_gpio_set(const struct shell *shell, size_t argc, char **argv,
 		return 0;
 	}
 
-	int gpio_index = args_info.arg[1].result.ivalue;
-	npmx_buck_gpio_config_t gpio_config = {
-		.inverted = args_info.arg[2].result.bvalue,
-	};
-
-	static const npmx_buck_gpio_t gpios[] = {
-		NPMX_BUCK_GPIO_0, NPMX_BUCK_GPIO_1, NPMX_BUCK_GPIO_2,
-		NPMX_BUCK_GPIO_3, NPMX_BUCK_GPIO_4,
-	};
-
-	if (!check_pin_configuration_correctness(shell, gpio_index)) {
+	npmx_buck_gpio_config_t gpio_config;
+	npmx_error_t err_code = gpio_config_get(buck_instance, &gpio_config);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "GPIO config");
 		return 0;
 	}
 
-	if (gpio_index == -1) {
-		gpio_config.gpio = NPMX_BUCK_GPIO_NC;
-	} else {
-		if ((gpio_index >= 0) && (gpio_index < ARRAY_SIZE(gpios))) {
-			gpio_config.gpio = gpios[gpio_index];
+	switch (config_type) {
+	case BUCK_GPIO_PARAM_INDEX:
+		if (args_info.arg[1].result.ivalue == -1) {
+			gpio_config.gpio = NPMX_BUCK_GPIO_NC;
 		} else {
-			shell_error(shell, "Error: wrong GPIO index.");
+			static const npmx_buck_gpio_t gpios[] = {
+				NPMX_BUCK_GPIO_0, NPMX_BUCK_GPIO_1, NPMX_BUCK_GPIO_2,
+				NPMX_BUCK_GPIO_3, NPMX_BUCK_GPIO_4,
+			};
+			if ((args_info.arg[1].result.ivalue >= 0) &&
+			    (args_info.arg[1].result.ivalue < ARRAY_SIZE(gpios))) {
+				gpio_config.gpio = gpios[args_info.arg[1].result.ivalue];
+			} else {
+				shell_error(shell, "Error: wrong GPIO index.");
+				return 0;
+			}
+		}
+		if (!check_pin_configuration_correctness(shell, gpio_config.gpio)) {
 			return 0;
 		}
+		break;
+	case BUCK_GPIO_PARAM_POLARITY:
+		gpio_config.inverted = args_info.arg[1].result.bvalue;
+		break;
+	default:
+		return 0;
 	}
 
-	npmx_error_t err_code = gpio_config_set(buck_instance, &gpio_config);
+	err_code = gpio_config_set(buck_instance, &gpio_config);
 	if (!check_error_code(shell, err_code)) {
 		print_set_error(shell, "GPIO config");
 		return 0;
 	}
 
-	print_success(shell, gpio_index, UNIT_TYPE_NONE);
+	int16_t *p_value = (config_type == BUCK_GPIO_PARAM_INDEX) ?
```

Comment:
Why pointer? Why not just a variable? Won't casting pointer to bool (`inverted` field) to a pointer to `uint16_t` cause any problems?
Same in LDSW commit

Reply:
Fixed

Reply:
It is still cast to pointer to `uint16_t` in  LDSW.

Reply:
Fixed there as well.